### ISSUE TITLE
Define Models

### ISSUE THREAD

**Issue by StanBarrows** [postcodes/api.postcodes.ch#13]
- [ ] Define Basic Models

https://github.com/postcodes/api.postcodes.ch/wiki/Models

any suggestions or improvements?

**Comment by nhedger** [comment]
Looks good to me. I only have one question regarding the Postcode Model. What would the `code_supplement` be ?

**Comment by StanBarrows** [comment]
HeyHey,

Thats when you multiple postcodes with the same 4digit code for a place.

![image](https://user-images.githubusercontent.com/10268813/87393244-28efb180-c5ae-11ea-9651-8960d6019a04.png)

**Comment by nhedger** [comment]
All clear now ! Thanks

**Comment by StanBarrows** [comment]
@nhedger 
We could also specify the spoken languages for a place. They already provide such a value but only one. I guess values are possible?

![image](https://user-images.githubusercontent.com/10268813/87393742-e8446800-c5ae-11ea-9ca3-16c52952b038.png)

Should we use a single column on each entry or adding a Language Model and attach the language for each entry?

`$table->json('languages')->nullable();`

**Comment by nhedger** [comment]
Indeed, there are cantons with multiple spoken languages. I think we can get away with a single column that holds an array of languages (in json). I'm happy giving the users access to the spoken languages but I'm not entirely sure keeping tabs on the percentages of speakers for a given language is in the scope of this project. At least, IMHO, it shouldn't for now.

**Comment by nhedger** [comment]
I see we have `deleted_at` columns on our Models. Are we going to keep track of deleted entries ? Should we ?

**Comment by StanBarrows** [comment]
i guess we dont need themn =)
